devnet-sdk/devstack/devtest/testing.go

```diff
@@ -0,0 +1,166 @@
+	// and inherits the context of the parent-test-scope.
+	Ctx() context.Context
+
+	// Parallel signals that this test is to be run in parallel with (and only with) other parallel tests.
+	Parallel()
+
+	Skip(args ...any)
+	Skipped() bool
+	Skipf(format string, args ...any)
+	SkipNow()
+
+	// This distinguishes the interface from other testing interfaces,
+	// such as the one used at package-level for shared system construction.
+	_TestOnly()
+}
+
+// This testing subset is sufficient for the require.Assertions to work.
+var _ require.TestingT = T(nil)
+
+// testingT implements the T interface by wrapping around a regular golang testing.T
+type testingT struct {
+	t      *testing.T
+	logger log.Logger
+	ctx    context.Context
+	req    *require.Assertions
+}
+
+func (t *testingT) Errorf(format string, args ...interface{}) {
+	t.t.Helper()
+	t.t.Errorf(format, args...)
+}
+
+func (t *testingT) FailNow() {
+	t.t.Helper()
+	t.t.FailNow()
+}
+
+func (t *testingT) TempDir() string {
+	return t.t.TempDir()
+}
+
+func (t *testingT) Cleanup(fn func()) {
+	t.t.Cleanup(fn)
+}
+
+func (t *testingT) Logf(format string, args ...any) {
+	t.t.Helper()
+	// Note: we do not use t.Log directly, to keep the log-formatting more consistent
+	t.logger.Info(fmt.Sprintf(format, args...))
+}
+
+func (t *testingT) Helper() {
+	t.t.Helper()
+}
+
+func (t *testingT) Name() string {
+	return t.t.Name()
+}
+
+func (t *testingT) Logger() log.Logger {
+	return t.logger
+}
+
+func (t *testingT) Ctx() context.Context {
+	return t.ctx
+}
+
+func (t *testingT) Require() *require.Assertions {
+	return t.req
+}
+
+func (t *testingT) Run(name string, fn func(T)) {
+	t.t.Run(name, func(subGoT *testing.T) {
+		ctx, cancel := context.WithCancel(t.ctx)
+		subGoT.Cleanup(cancel)
+		subT := &testingT{
+			t:      subGoT,
+			logger: t.logger.New("subtest", name),
+			ctx:    ctx,
+		}
+		fn(subT)
+	})
+}
+
+func (t *testingT) Parallel() {
+	t.logger.Info("Running test in parallel")
+	t.t.Parallel()
+}
+
+func (t *testingT) Skip(args ...any) {
+	t.t.Helper()
+	t.t.Skip(args...)
+}
+
+func (t *testingT) Skipped() bool {
+	t.t.Helper()
+	return t.t.Skipped()
+}
+
+func (t *testingT) Skipf(format string, args ...any) {
+	t.t.Helper()
+	t.t.Skipf(format, args...)
+}
+
+func (t *testingT) SkipNow() {
+	t.t.Helper()
+	t.t.SkipNow()
+}
+
+func (t *testingT) _TestOnly() {
+	panic("do not use - this method only forces the interface to be unique")
+}
+
+var _ T = (*testingT)(nil)
+
+// SerialT wraps around a test-logger and turns it into a T for devstack testing.
+func SerialT(t *testing.T) T {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	logger := testlog.Logger(t, log.LevelInfo)
+	out := &testingT{
+		t:      t,
+		logger: logger,
+		ctx:    ctx,
+	}
+	out.req = require.New(out)
+	return out
+}
+
+// ParallelT creates a T interface with parallel testing enabled by default
+func ParallelT(t *testing.T) T {
+	out := SerialT(t)
+	out.Parallel()
```

Comment:
```suggestion
	t.Parallel()
```

So we don't need `Parallel` on the interface.

Reply:
See other comment